test/jdk/java/util/TimeZone/ZoneOffsetRoundTripTest.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright (c) 2022, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.time.ZoneId;
+import java.util.TimeZone;
+
+import org.testng.annotations.Test;
+import org.testng.annotations.DataProvider;
+import static org.testng.Assert.assertEquals;
+
+/**
+ * @test
+ * @bug 8285844
+ * @summary Checks round-trips between TimeZone and ZoneOffset are consistent
+ * @run testng ZoneOffsetRoundTripTest
+ */
+@Test
+public class ZoneOffsetRoundTripTest {
+
+    @DataProvider
+    private Object[][] testZoneOffsets() {
+        return new Object[][] {
+                {ZoneId.of("Z"), 0},
```

Comment:
I know, `ZoneId.of()` should parse this as a `ZoneOffset` and return a `ZoneOffset` instance, but maybe add also the other string variants with prefix (`ZoneId.of("UTC+00:00:01")` or `ZoneId.of("GMT+00:00:01")` as data items. Maybe also use `ZoneOffset.of()` for the plain zones to be explicit.

Reply:
Added them except "UTC+...", as it is not recognizable as a Custom ID.

Reply:
Can the test cover `UT` prefix as well? (This is another valid prefix in `ZoneId`)

If this PR isn't meant to work with UTC prefix, can a test be added that proves it does *not* work.

ie. all these are valid in `ZoneId` - "Z", "UTC", "GMT", "UT", "UTC+01:00", "GMT+01:00", "UT+01:00" - and all should have some form of associated test.

Reply:
I tried it out: If you create a ZoneId with prefixes "UT" and "UTC", the fail to convert to TimeZone. Smae happens if you use this as String name in `TimeZone#getTimeZone(String)`. This is another bug / missing feature! It does not work with or without this PR.

Reply:
The code intended to allow only offset-style ids that start with `GMT` (which is compatible with TimeZone's CustomID). Now I made changes to allow all offset-style zone ids.